### PR TITLE
Some tweaks to logging in kdd

### PR DESCRIPTION
**Pull request by caseydavenport**

### CODE REVIEW

File: lib/backend/k8s/k8s.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ type KubeClient struct {
 
 	// Clients for interacting with Calico resources.
 	ipPoolClient api.Client
-	nodeClient api.Client
+	nodeClient   api.Client
```
Comment (djosborne) [inline lib/backend/k8s/k8s.go]: ...okay

File: lib/backend/k8s/k8s.go [hunk 2]
```diff
@@ -223,6 +223,7 @@ func (c *KubeClient) ensureClusterType() (bool, error) {
 		}
 		value = existingValue
 	}
+	log.Debugf("Setting ClusterType to '%s'", value)
```
Comment (djosborne) [inline lib/backend/k8s/k8s.go]: Would now be a good time for us to start making use of [logrus' `withField`](https://github.com/sirupsen/logrus#example)?

```
log.WithField("ClusterType", value).Debug("Setting ClusterType")
```

File: lib/backend/k8s/resources/node_conversion.go [hunk 3]
```diff
@@ -37,7 +39,7 @@ func K8sNodeToCalico(node *kapiv1.Node) (*model.KVPair, error) {
 	ip := net.ParseIP(nodeIP)
 	log.Debugf("Node IP is %s", ip)
```
Comment (djosborne) [inline lib/backend/k8s/resources/node_conversion.go]: Maybe should this go in an `else` or be removed

File: lib/backend/k8s/k8s.go (hunk 1 above)
Comment (caseydavenport) [inline lib/backend/k8s/k8s.go]: `gofmt` fixed this.

**Comment by caseydavenport** [comment]
@djosborne any chance we could get this in? I'd like to have the log level tweaks.

File: lib/backend/k8s/k8s.go [hunk 4]
```diff
@@ -278,6 +279,7 @@ func (c *KubeClient) Syncer(callbacks api.SyncerCallbacks) api.Syncer {
 
 // Create an entry in the datastore.  This errors if the entry already exists.
 func (c *KubeClient) Create(d *model.KVPair) (*model.KVPair, error) {
+	log.Debugf("Performing 'Create' for %+v", d)
```
Comment (robbrockbank) [inline lib/backend/k8s/k8s.go]: Better to use WithField() here

File: lib/backend/k8s/syncer.go [hunk 5]
```diff
@@ -274,7 +275,7 @@ func (syn *kubeSyncer) readFromKubernetesAPI() {
 			log.Debugf("Incoming NetworkPolicy watch event. Type=%s", event.Type)
 			if needsResync = syn.eventTriggersResync(event); needsResync {
 				// We need to resync.  Break out into the sync loop.
-				log.Warn("Event triggered resync: %+v", event)
+				log.Warnf("Event triggered resync: %+v", event)
```
Comment (robbrockbank) [inline lib/backend/k8s/syncer.go]: Again, these should really use WithField() I think.

Comment (caseydavenport) [inline lib/backend/k8s/syncer.go]: Does WithField give you %+v formatting?

Comment (robbrockbank) [inline lib/backend/k8s/syncer.go]: Ah I see - no idea TBH.  Probably not, in which case LGTM.

Comment (robbrockbank) [inline lib/backend/k8s/syncer.go]: I'm not sure - but always doing the formatting when we're not logging doesn't feel right to me.  I guess it won't add much overhead so I'd be ok with merging this.

Would it be better though to add a String() method to the KVPair and then use WithField()

Comment (caseydavenport) [inline lib/backend/k8s/syncer.go]: I'm not too fussed about this one.

Looking at the logging code, the String formatting doesn't actually get done unless the correct log level is set, so it won't even have a perf impact.  Plus, this is a low-frequency warning log.

Comment (robbrockbank) [inline lib/backend/k8s/syncer.go]: Oh, of course!   Yes - in that case LGTM.

**Pull request merged**